kafka-rest/src/main/java/io/confluent/kafkarest/ProducerMetrics.java

```diff
@@ -0,0 +1,426 @@
+    private final ProduceMetricMBean bean;
+    private final Sensor sensor;
+
+    public SensorBuilder(ProduceMetricMBean bean, String jmxPrefix, String name) {
+      this.bean = bean;
+      this.sensor =
+          bean.sensor(String.join(":", jmxPrefix, ProducerMetricsRegistry.GROUP_NAME, name));
+    }
+
+    public SensorBuilder addAvg(String name, String doc) {
+      MetricName metricName = getMetricName(name, doc);
+      sensor.add(metricName, MeasuredStatSupplier.avg());
+      return this;
+    }
+
+    public SensorBuilder addRate(String name, String doc) {
+      MetricName metricName = getMetricName(name, doc);
+      sensor.add(metricName, MeasuredStatSupplier.rate());
+      return this;
+    }
+
+    public SensorBuilder addMax(String name, String doc) {
+      MetricName metricName = getMetricName(name, doc);
+      sensor.add(metricName, MeasuredStatSupplier.max());
+      return this;
+    }
+
+    public SensorBuilder addWindowedCount(String name, String doc) {
+      MetricName metricName = getMetricName(name, doc);
+      sensor.add(metricName, MeasuredStatSupplier.windowedCount());
+      return this;
+    }
+
+    public SensorBuilder addPercentiles(
+        String name, Map<String, Double> suffixPercentiles, String doc) {
+      Map<MetricName, Double> namePercentiles =
+          suffixPercentiles.entrySet().stream()
+              .map(
+                  suffixPercentile ->
+                      new AbstractMap.SimpleImmutableEntry<MetricName, Double>(
+                          getMetricName(name + suffixPercentile.getKey(), doc),
+                          suffixPercentile.getValue()))
+              .collect(Collectors.toMap(e -> e.getKey(), e -> e.getValue()));
+
+      sensor.add(MeasuredStatSupplier.percentiles(namePercentiles));
+      return this;
+    }
+
+    public Sensor build() {
+      return sensor;
+    }
+
+    private MetricName getMetricName(String name, String doc) {
+      MetricName metricName =
+          bean.metricName(
+              new MetricNameTemplate(
+                  name, ProducerMetricsRegistry.GROUP_NAME, doc, Collections.emptySet()));
+      return metricName;
+    }
+  }
+
+  public static class MeasuredStatSupplier {
+
+    public static MeasurableStat avg() {
+      return new Avg();
+    }
+
+    public static MeasurableStat rate() {
+      return new Rate();
+    }
+
+    public static MeasurableStat max() {
+      return new Max();
+    }
+
+    public static MeasurableStat windowedCount() {
+      return new WindowedCount();
+    }
+
+    public static CompoundStat percentiles(Map<MetricName, Double> percentiles) {
+      return new Percentiles(
+          1000,
+          250,
+          Percentiles.BucketSizing.CONSTANT,
+          percentiles.entrySet().stream()
+              .map(
+                  namePercentile ->
+                      new Percentile(namePercentile.getKey(), namePercentile.getValue()))
+              .toArray(Percentile[]::new));
+    }
+  }
```

Comment:
This means that the max latency we can track is just `250`, right? That seems way too low to me. I think we also track latency as an integer number, which means that we actually have almost no precision loss for the percentile histogram. I'd suggest bumping the `sizeInBytes` a bit (as this would be a global metric) and then bumping significantly the max latency that we can track.

Reply:
I'd pulled the values from fetch metrics in ce-kafka but looks like i was looking in the wrong place. Have updated with sensible 30s max now.